pkg/cmd/roachtest/tests/versionupgrade.go

```diff
@@ -287,3 +290,69 @@ for i in 1 2 3 4; do
 done
 `)
 }
+
+func registerHTTPRestart(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Name:             "http-restart/mixed-version",
+		Owner:            registry.OwnerTestEng,
+		Cluster:          r.MakeClusterSpec(4),
+		CompatibleClouds: registry.AllClouds,
+		Suites:           registry.Suites(registry.MixedVersion, registry.Nightly),
+		Randomized:       true,
+		Run:              runHTTPRestart,
+		Timeout:          1 * time.Hour,
+	})
+}
+
+func runHTTPRestart(ctx context.Context, t test.Test, c cluster.Cluster) {
+	mvt := mixedversion.NewTest(ctx, t, t.L(), c,
+		c.CRDBNodes(),
+		mixedversion.MinimumSupportedVersion("v24.2.0"),
+		mixedversion.MinUpgrades(2),
+		mixedversion.EnabledDeploymentModes(mixedversion.SeparateProcessDeployment),
+	)
+
+	httpCall := func(ctx context.Context, node int, l *logger.Logger) {
+		logEvery := roachtestutil.Every(1 * time.Second)
+		adminURLs, err := c.ExternalAdminUIAddr(ctx, l, c.Node(node))
+		if err != nil {
+			t.Fatal(err)
+		}
+		url := "https://" + adminURLs[0] + "/ts/query"
+		l.Printf("Sending requests to %s", url)
+		client := roachtestutil.DefaultHTTPClient(c, l)
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			// Copied from elsewhere, actual HTTP req shouldn't matter
+			var response tspb.TimeSeriesQueryResponse
+			request := tspb.TimeSeriesQueryRequest{
+				StartNanos: timeutil.Now().UnixNano() - 10*time.Second.Nanoseconds(),
+				EndNanos:   timeutil.Now().UnixNano(),
+				// Ask for 10s intervals.
+				SampleNanos: (10 * time.Second).Nanoseconds(),
+				Queries: []tspb.Query{{
+					Name:             "cr.node.sql.service.latency-p90",
+					SourceAggregator: tspb.TimeSeriesQueryAggregator_MAX.Enum(),
+				}},
+			}
+			if err := client.PostProtobuf(ctx, url, &request, &response); err != nil {
+				if logEvery.ShouldLog() {
+					l.Printf("Error posting protobuf: %s", err)
+				}
+			}
+		}
+	}
+
+	for i := 1; i <= 4; i++ {
```

Comment:
I think we should use `for _, node := range c.CRDBNodes() {...`  instead of a hardcoded size

Reply:
Yep, hardcoding should be eschewed because it makes the test configuration brittle, i.e., difficult to refactor.